internal/service/elasticache/cluster.go

```diff
@@ -504,7 +505,7 @@ func resourceClusterRead(ctx context.Context, d *schema.ResourceData, meta any)
 
 	c, err := findCacheClusterWithNodeInfoByID(ctx, conn, d.Id())
 
-	if !d.IsNewResource() && tfresource.NotFound(err) {
+	if !d.IsNewResource() && retry.NotFound(err) {
```

Comment:
I'm thinking now it's probably best to keep this as `tfresource.NotFound`, at least until we are sure nothing uses the SDKv2 `helper/retry` package.